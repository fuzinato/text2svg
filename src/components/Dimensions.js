import React from 'react'
import Wrapper from './Wrapper';
import { string, number, func } from 'prop-types';

const Dimensions = (props) => {
  function handleChange(e) {
    const val = e.target.value || 0
    const state = { [props.measures]: val }
    console.log(state)
    props.onChange(state)
  }
  return (
    <Wrapper title={props.title}>
      <input
        type="number"
        placeholder={props.title}
        defaultValue={props.text}
        onChange={handleChange} />
    </Wrapper>
  )
}

Dimensions.propTypes = {
  onChange: func.isRequired,
  title: string.isRequired,
  measures: string.isRequired,
  text: number,
};

export default Dimensions
