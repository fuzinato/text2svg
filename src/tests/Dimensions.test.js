import React from 'react';
import Dimensions from '../components/Dimensions';
import renderer from 'react-test-renderer';

describe('Dimensions ', () => {

  test('should have correct shapshot', () => {
    const handleChange = () => {}
    const component = renderer.create(
      <Dimensions onChange={handleChange} title="Dummy Title" text={51} measures="foobar" />
      
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});