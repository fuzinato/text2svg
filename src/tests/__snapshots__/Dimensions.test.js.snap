// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dimensions  should have correct shapshot 1`] = `
<div
  className="input-section undefined"
>
  <span>
    Dummy Title
  </span>
  <input
    defaultValue={51}
    onChange={[Function]}
    placeholder="Dummy Title"
    type="number"
  />
</div>
`;
